proposals/0000-things-to-improve.md

```diff
@@ -0,0 +1,69 @@
+<!-- {% raw %} -->
+
+# List of Things to Improve in the HLSL Language
```

Comment:
I'm wondering if a proposal document is the right way to capture this?  Any reason not to use issues for each thing?  Then a proposal document can be written with a concrete proposal on addressing the thing-that-needs-to-be-improved.

Reply:
I've created this list based on a @llvm-beanz's suggestion that it would be helpful to have one place that lists all of the not-so-nice features and obscure constructs in the language. We can then refer to it when designing a new language version. Issues would probably too as long as they are labeled such that they're easy to find.